docs/adr/0001-daily-active-use-server-side-metrics-glean.md

```diff
@@ -0,0 +1,113 @@
+
+## Decision Drivers
+
+1. Simplicity of implementation, not only for internal metric emission, but for processing and querying.
+2. Reduction of complexity and load on team to process and make sense of data. Ideally there is existing infrastructure to reduce the possible significant work to calculate DAU.
+3. Glean is an internal tool successfully used by many teams at Mozilla.
+4. The Glean team has extensive experience and can provide considerable support in establishing application metrics.
+5. Extensibility: ability to expand scope of metrics to other measurements and apply methodology to other services in Sync ecosystem.
+
+## Considered Options
+
+* A. Glean
+* B. StatsD and Grafana
+* C. Sql/Redash
+
+## Decision Outcome
+
+Chosen option:
+
+* A. "Glean for server-side measurement of DAU"
+
+The use of Glean appears the best choice for measuring internal DAU metrics.  It meets our requirements and provides us with needed support on the data processing side. It also provides considerable support from the Glean team to implement this in a thoughtful manner.  There are some challenges with this implementation (more below), namely in this being a greenfield attempt at Rust server-side metrics, however the pros outweigh the cons. Other metrics implementations like StatsD and Grafana cannot be easily used to measure and aggregate this data. Additionally, it adds considerable overhead in determining how to query the data and reconcile ping emissions of Sync events. Having dedicated organizational support means we establish best practices.
+
+## Pros and Cons of the Options
+
+### A. Glean
+
+Glean is a widely used tool at Mozilla and provides us with a solution to the given issue and possible extensibility in the future. Not without some challenges in initial implementation, but the potential for positive impact is high.
+
+#### Pros
+
+* Satisfaction of requirement to measure internal DAU metrics.
+* Core of Glean's purpose is to measure user interactions and the rich metadata that accompanies it.
+* Capacity for future expansion of application metrics within Sync beyond DAU.
+* Prepares for implementation of same measurements in autopush, also using Glean.
+* Easier to query.
+* Use of standardized Mozilla tooling.
+* Establishment of team knowledge of using Glean.
+* Establishment of server-side Rust best practices, leading to easier development for backend Rust applications.
+* Transparency in data review process with consideration made to minimum collection of data.
+
+#### Cons
+
+* The `glean_parser` tool used by other Glean applications is currently not supported for Rust. Furthermore, client applications can use the Glean SDK and this is also not an option for us.
```

Comment:
I think we should be more clear here. There is no immediate support for the traditional `glean_parser` code generation for Rust. Rust is more than capable of generating output that is parseable by Glean. Ideally, the Rust code would be generated from the XML declaration file using a combination of jinja2 templates and other tooling which the Sync team is unfamiliar with. The output of which would still need to be validated by the Glean team. (I'll note that if we were to generate the `glean_parser` code generation element, I presume that the Glean team would also want to review that.) 

(Somewhat personally, rust, like java, does not take well to the idea of "automatically generated and inserted code", so the code would need to be pre-generated and made part of the repo before it could be tested and deployed. This would be so that all the additional tooling, such as CVE audit checks, formatting checks, and other code quality checks are performed on the entire code base.)

Reply:
There's one big "con" here, and that's that we have to write something -- we don't get Glean for free via glean_parser. SDK is not part of the considered option here, so that sentence should go entirely.

Re: clarity, I don't think the first sentence is substantially different from jr's second sentence. We just need to state the big drawback of this option; other options aren't relevant to the discussion here.